### ISSUE TITLE
Why Lua?

### ISSUE THREAD

**Issue by matteoturini** [K4rakara/freshfetch#22]
Why is lua necessary? Isn't it possible to not use Lua?

**Comment by K4rakara** [comment]
Lua is used to allow users to customize freshfetch, rather than make their own fetch just for a different look. The API isn't very user friendly as of writing this, but it literally lets you do anything you want, within the limits of ANSI escape codes.

**Comment by matteoturini** [comment]
Can't these ANSI codes be implemented using string replacing or compiling
the syntax to a text file? Or at least giving an option to build it without
Lua? I feel like this is unnecessary bloat for people that don't use it.

Em seg., 3 de mai. de 2021 20:03, Jack ***@***.***> escreveu:

**Comment by matteoturini** [comment]
+ from what I understood, no dynamic features are currently being used. It is better to make a "compiler" for the ANSI escape code features that are required for customization.

**Comment by K4rakara** [comment]
What Lua also allows is for users to fetch their own information.

I may make a cfg flag that removes lua when I rework the codebase, but it wouldn't be the default.

**Comment by matteoturini** [comment]
> fetch their own information.

Why not make a rust config then? you remove the lua (bloat) and allow
customization through editing the source.

Em ter., 4 de mai. de 2021 às 14:33, Jack ***@***.***>
escreveu:

**Comment by K4rakara** [comment]
I don't believe in the suckless mentality of forcing users to recompile their programs to configure them.

I'm not _against_ having a rust config, but if i did it it would be dynamically loaded or something.

**Comment by matteoturini** [comment]
Yeah, why not bundle rustc and compile config with a flag.

**Comment by Speykious** [comment]
@K4rakara I'm 2 months late in the matter, but what about making it configurable through a TOML file? It's kind of a standard for Rust programs and it works really well.
Take a look at [Starship](https://starship.rs) for example. It's an incredibly extensible prompt also written in Rust, and [it's configured with TOML](https://starship.rs/config/).

(Maybe I can even try to code that in, hmmm)

**Comment by matteoturini** [comment]
Lua is too complex for a simple project like this. I agree that TOML is a
good, widespread format. If the API isn't good or user friendly, why is it
there?

Em qua., 14 de jul. de 2021 02:19, Speykious ***@***.***>
escreveu:

**Comment by matteoturini** [comment]
Glad you thought about it. Good luck with the rewrite!
